packages/ui-library/src/components/transaction-receipt/index.tsx

```diff
@@ -0,0 +1,306 @@
+        </Typography>
+      </StyledSectionContent>
+      <StyledSectionContent>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="From address" />
+        </Typography>
+        <Typography variant="body" fontWeight="bold" color={colors[mode].typography.typo2}>
+          {transaction.owner}
+        </Typography>
+      </StyledSectionContent>
+      <StyledSectionContent>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="Spender" />
+        </Typography>
+        <Typography variant="body" fontWeight="bold" color={colors[mode].typography.typo2}>
+          {transaction.spender}
+        </Typography>
+      </StyledSectionContent>
+    </>
+  );
+};
+
+const ERC20TransferTransactionReceipt = ({ transaction }: { transaction: ERC20TransferEvent }) => {
+  const {
+    palette: { mode },
+    spacing,
+  } = useTheme();
+  return (
+    <>
+      <StyledSectionContent>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="Amount sent" />
+        </Typography>
+        <Typography
+          variant="body"
+          fontWeight="bold"
+          sx={{ display: 'flex', alignItems: 'center', gap: spacing(2) }}
+          color={colors[mode].typography.typo2}
+        >
+          {transaction.token.icon}
+          {transaction.amount.amountInUnits} {transaction.amount.amountInUSD && `($${transaction.amount.amountInUSD})`}
+        </Typography>
+      </StyledSectionContent>
+      <StyledSectionContent>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="From address" />
+        </Typography>
+        <Typography variant="body" fontWeight="bold" color={colors[mode].typography.typo2}>
+          {transaction.from}
+        </Typography>
+      </StyledSectionContent>
+      <StyledSectionContent>
+        <Typography variant="bodySmall" color={colors[mode].typography.typo3}>
+          <FormattedMessage description="TransactionReceipt-transactionAmountSent" defaultMessage="To address" />
+        </Typography>
+        <Typography variant="body" fontWeight="bold" color={colors[mode].typography.typo2}>
+          {transaction.to}
+        </Typography>
+      </StyledSectionContent>
+    </>
+  );
+};
+
+const buildTransactionReceiptForEvent = (transaction: TransactionEvent) => {
+  switch (transaction.type) {
+    case TransactionEventTypes.ERC20_APPROVAL:
+      return <ERC20ApprovalTransactionReceipt transaction={transaction} />;
+      break;
```

Comment:
We could safely remove break statements due to early returns

Reply:
True